ipfs/ipfs.go

```diff
@@ -60,12 +59,8 @@ func (ipfs *IPFS) Add(name string, reader io.Reader) (*Response, error) {
 	if res.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("bad status: %s", res.Status)
 	}
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, err
-	}
 	var response Response
-	if err := json.Unmarshal(body, &response); err != nil {
+	if err := json.NewDecoder(res.Body).Decode(&response); err != nil {
```

Comment:
no need for _if_ we can return this error directly

Reply:
somethin like this:

```go
var response Response
err := json.NewDecoder(res.Body).Decode(&response)
return &response, err
```

Reply:
But then between `err := json.NewDecoder(res.Body).Decode(&response)` and `return &response, err` if we add an instruction we might miss an error. I like to immediately check the error or return all the result directly